### PR TITLE
overwrite `addBatchSinglePlace` and `addBatchSinglePlaceNotNull` in `AggregateFunctionCountNotNullUnary`

### PR DESCRIPTION
**Pull request by windtalker**
### What problem does this PR solve?

Issue Number: ref #9146

Problem Summary:

### What is changed and how it works?
`AggregateFunctionCountNotNullUnary` does not overwrite `addBatchSinglePlace` and `addBatchSinglePlaceNotNull`, so the code path will be 
https://github.com/pingcap/tiflash/blob/08535236b8f3d6d41656f22ebb8c2334f3081d86/dbms/src/AggregateFunctions/IAggregateFunction.h#L267-L289

In fact, `AggregateFunctionCountNotNullUnary` can be optimized by count the element number in batch like `AggregateFunctionCount` by overwrite `addBatchSinglePlace` and `addBatchSinglePlaceNotNull` 
```commit-message

```

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by yibin87 — COMMENTED** [review]
What're the differences between these two functions? Only the way that the nullmap is passed?

**Comment by windtalker** [comment]
> What're the differences between these two functions? Only the way that the nullmap is passed?

`addBatchSinglePlace` is the public function that will be called from `Aggregator` directly, `addBatchSinglePlaceNotNull` is kind of a protect function that usually be called inside AggregateFunctionXXX.

File: dbms/src/AggregateFunctions/AggregateFunctionCount.h [hunk 1]
```diff
@@ -131,6 +131,47 @@ class AggregateFunctionCountNotNullUnary final
...
+        if (if_argument_pos >= 0)
+        {
+            const auto & flags = assert_cast<const ColumnUInt8 &>(*columns[if_argument_pos]).getData();
+            data(place).count
+                += countBytesInFilterWithNull(flags, nc.getNullMapData().data(), start_offset, batch_size);
+        }
+        else
+        {
+            data(place).count += batch_size - countBytesInFilter(nc.getNullMapData().data(), start_offset, batch_size);
+        }
```
Comment (gengliqi) [inline dbms/src/AggregateFunctions/AggregateFunctionCount.h:152]: How about calling `addBatchSinglePlaceNotNull` here directly?

Comment (windtalker) [reply dbms/src/AggregateFunctions/AggregateFunctionCount.h]: Good idea.

**Review by gengliqi — APPROVED** [review]

**Review by yibin87 — APPROVED** [review]
LGTM

**Pull request merged**